packages/server/src/sdk/app/rows/search/internal/sqs.ts

```diff
@@ -70,13 +97,18 @@ function buildInternalFieldList(
     })
     return junctionFields
   }
-  fieldList = fieldList.concat(
-    PROTECTED_INTERNAL_COLUMNS.map(col => `${table._id}.${col}`)
-  )
-  for (let key of Object.keys(table.schema)) {
+  if (sdk.tables.isTable(source)) {
+    for (const key of PROTECTED_INTERNAL_COLUMNS) {
+      if (allowedFields && !allowedFields.includes(key)) {
```

Comment:
Could this be done with a filter instead - `PROTECTED_INTERNAL_COLUMNS.filter(key => !allowedFields?.includes(key)).map(key => ${table._id}.${key})`

Reply:
It certainly could, but this felt a little cleaner. Happy to change it if you want but don't think the benefit is that large.

Reply:
I generally find the nature of `.filter` and `.map` a bit more readable than a for loop since its clear what its doing, making a new sub-array rather than performing any explicit actions, but very much preference!